File: easy/1758_minimum_changes_to_make_alternating_binary_string/solution.py
class Solution:
    def minOperations(self, s: str) -> int:
        if len(s) == 1:
            return 0
        if len(s) == 2:
            return int(s[0] == s[1])

        zeros = ones = 0
        zero = True

        for i in range(len(s)):
            if (zero and s[i] != "0") or (not zero and s[i] != "1"):
                zeros += 1
            if (zero and s[i] != "1") or (not zero and s[i] != "0"):
                ones += 1

            zero = not zero

        return min(zeros, ones)

File: easy/1758_minimum_changes_to_make_alternating_binary_string/solution.cpp
#include <string>

class Solution {
public:
  int minOperations(std::string s) {
    if (s.size() == 1)
      return 0;
    if (s.size() == 2)
      return s[0] == s[1];

    int changesCount = 0;
    int curr = 0;

    for (size_t i = 0; i < s.size(); ++i) {
      if ((curr == 0 && s[i] != '0') || (curr == 1 && s[i] != '1'))
        ++changesCount;

      curr = !curr;
    }

    return std::min(changesCount, (int)s.size() - changesCount);
  }
};
